mlflow/azureml/__init__.py

```diff
@@ -380,15 +382,41 @@ def deploy(
             _copy_file_or_tree(src=absolute_model_path, dst=model_directory_path),
         )
 
-        registered_model = AzureModel.register(
-            workspace=workspace, model_path=tmp_model_path, model_name=model_name, tags=tags
-        )
+        registered_model = None
+
+        try:
+            if model_uri.startswith("models:/"):
+                m_name = model_uri.split("/")[-2]
+                m_version = int(model_uri.split("/")[-1])
+                m_id = "{}:{}".format(m_name, m_version)
+                registered_model = AzureModel(workspace, id=m_id)
+
+                _logger.info("Found registered model in AzureML with ID '%s'", m_id)
+            elif model_uri.startswith("runs:/") and get_tracking_uri().startswith("azureml"):
+                m = mlflow_register_model(model_uri, model_name)
+                m_id = "{}:{}".format(m.name, m.version)
+                registered_model = AzureModel(workspace, id=m_id)
```

Comment:
My delete seems to have confused github.

We should try to scope down the try catch to only wrap this call with m_id

Reply:
Discussed offline.